fuel-txpool/src/txpool.rs

```diff
@@ -145,11 +150,53 @@ impl TxPool {
         Ok(())
     }
 
+    #[cfg(feature = "p2p")]
+    pub async fn insert_with_broadcast(
+        txpool: &RwLock<Self>,
+        db: &dyn TxPoolDb,
+        tx_status_sender: broadcast::Sender<TxStatusBroadcast>,
+        network_sender: mpsc::Sender<P2pRequestEvent>,
+        txs: Vec<ArcTx>,
+    ) -> Vec<anyhow::Result<Vec<ArcTx>>> {
+        let mut res = Vec::new();
+        for tx in txs.iter() {
+            let mut pool = txpool.write().await;
+            res.push(pool.insert_inner(tx.clone(), db).await)
+        }
+        for (ret, tx) in res.iter().zip(txs.into_iter()) {
+            match ret {
+                Ok(removed) => {
+                    for removed in removed {
+                        let _ = tx_status_sender.send(TxStatusBroadcast {
+                            tx: removed.clone(),
+                            status: TxStatus::SqueezedOut {
+                                reason: Error::Removed,
+                            },
+                        });
+                    }
+                    let _ = tx_status_sender.send(TxStatusBroadcast {
+                        tx: tx.clone(),
+                        status: TxStatus::Submitted,
+                    });
+                    let _ = network_sender
```

Comment:
Maybe rather than complicating with a closure, this method could just call the one below, iterate over the `Vec<Result<Arc<Transaction>>>` return type and then trigger a broadcast for each successful one.

Reply:
> Maybe rather than complicating with a closure, this method could just call the one below, iterate over the `Vec<Result<Arc<Transaction>>>` return type and then trigger a broadcast for each successful one.

We originally proposed this but we changed it in favour of a more "performant" approach, at the cost of the extra code. But I think this approach is the simplest and cleanest, and I prefer the cleaner approach to the performant approach in this case. 

The draw back is:
- Double the number of iterations
 
But the benefits are:
- No extra method or code duplication
- Easier to read and reason about code

Reply:
We should probably atleast leave a note in the future though in case in future txpool optimizations we need areas to improve

Reply:
Draft PR based on this branch: https://github.com/FuelLabs/fuel-core/pull/670